os/mm/mm_heap/mm_heapinfo_parse_heap.c

```diff
@@ -110,6 +115,34 @@ void heapinfo_parse_heap(FAR struct mm_heap_s *heap, int mode, pid_t pid)
 
 	ASSERT(mm_check_heap_corruption(heap) == OK);
 
+#ifdef CONFIG_WATCHDOG
+	bool is_watchdog_running = false;
+	struct watchdog_status_s wd_status;
+	int wd_fd = open(CONFIG_WATCHDOG_DEVPATH, O_RDONLY);
+	if (wd_fd >= 0) {
+		if (ioctl(wd_fd, WDIOC_GETSTATUS, (unsigned long)&wd_status) != OK) {
+			is_watchdog_running = wd_status.flags & WDFLAGS_ACTIVE;
+		} else {
+			mfdbg("Fail to get watchdog state, errno %d\n", get_errno());
+			close(wd_fd);
+#ifdef CONFIG_MM_ASSERT_ON_FAIL
+			mfdbg("Watchdog can reboot during the printing heap usage log.\n");
+#else
+			mfdbg("It doesn't print the heap usage dump.\n");
+			return;
+#endif
+		}
+	} else {
+		mfdbg("Fail to open %s, errno %d\n", CONFIG_WATCHDOG_DEVPATH, get_errno());
+#ifdef CONFIG_MM_ASSERT_ON_FAIL
+		mfdbg("Watchdog can reboot during the printing heap usage log.\n");
+#else
+		mfdbg("It doesn't print the heap usage dump.\n");
+		return;
+#endif
+	}
+#endif
```

Comment:
Let's make it as a separate function.

Reply:
thanks you for suggestion.

I separated checking wd fd and checking wd state.